app/jobs/process_delius_data_job.rb

```diff
@@ -0,0 +1,67 @@
+class ProcessDeliusDataJob < ApplicationJob
+  queue_as :default
+
+  def perform(nomis_offender_id)
+    DeliusData.transaction do
+      DeliusImportError.where(nomis_offender_id: nomis_offender_id).destroy_all
+
+      delius_data = DeliusData.where(noms_no: nomis_offender_id)
+      if delius_data.count > 1
+        DeliusImportError.create! nomis_offender_id: nomis_offender_id,
+                                  error_type: DeliusImportError::DUPLICATE_CRN
+      else
+        process_record(delius_data.first)
+      end
+    end
+  end
+
+private
+
+  def process_record(delius_record)
+    case_information = CaseInformation.find_or_initialize_by(
```

Comment:
I think, unless we match on noms_no _and_ CRN we might not catch duplicates, that is more than one offender being given the noms_no in Delius.  This would just overwrite possibly the wrong offender I think.

Reply:
Yes that thought crossed my mind in the last couple of days too.